### DISCUSSION TITLE
Help avoiding usage of unserializable parameters in proxy services.

### DISCUSSION THREAD

**Discussion by jalooc** [aklinker1/webext-core#76] [Ideas]
The proxy services functioning is based on `postMessage()`, which only works with serializable data. It would be nice touch to prevent devs from passing any non-serializable data, like functions or classes (e.g. `CustomError` thrown in background script will fail the `instanceof CustomError` check in a content script), to avoid wtf moments when it doesn't reach the other end of the communication channel, being silently filtered out. I thought this could be done with TS by making sure the methods of the object passed to `defineProxyService` don't have parameters or return values of illegal types.

**Comment by aklinker1** [comment]
Fixed by https://github.com/aklinker1/webext-core/pull/79, released in https://github.com/aklinker1/webext-core/releases/tag/messaging-v2.0.3
